garage/tf/models/base.py

```diff
@@ -0,0 +1,51 @@
+"""This file contains the tautology for garage models."""
+import tensorflow as tf
+from tensorflow.keras.models import Model as KerasModel
+
+from garage.misc.overrides import overrides
+from garage.tf.core import Parameterized
+
+
+class Model(KerasModel, Parameterized):
+    """
+    This is a tautology for all the garage models.
+
+    A model define a neural network model to build a graph for
+    other primitivs (e.g. GaussianMLPPolicy and GaussianMLPBaseline).
+    A model can be built by assembling different models. For example,
+    to define a VAEGaussianMLP Model, we need to build a VAE model
+    and a GaussianMLP Model. Then, we can assemble them in the
+    build_model function.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a model."""
+        super(Model, self).__init__(*args, **kwargs)
+        Parameterized.__init__(self)
+
+    def build_model(self, inputs=None, reuse=tf.AUTO_REUSE):
```

Comment:
This is called `call()` in the keras API

https://www.tensorflow.org/api_docs/python/tf/keras/models/Model (see bullet point 2)